src/storage/hummock_trace/src/replay/mod.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod runner;
+mod worker;
+
+use std::ops::Bound;
+
+#[cfg(test)]
+use mockall::automock;
+use risingwave_pb::meta::subscribe_response::{Info, Operation as RespOperation};
+pub use runner::*;
+pub(crate) use worker::*;
+
+use crate::error::Result;
+use crate::{Record, TraceReadOptions, TraceWriteOptions};
+
+type ReplayGroup = Record;
+
+type WorkerResponse = ();
+
+#[derive(Debug)]
+pub(crate) enum ReplayRequest {
+    Task(ReplayGroup),
+    Fin,
+}
+
+#[derive(PartialEq, Eq, Hash, Debug, Clone)]
+pub(crate) enum WorkerId {
+    Local(u64),
+    OneShot(u64),
+}
+
+#[cfg_attr(test, automock)]
+#[async_trait::async_trait]
+pub trait Replayable: Send + Sync {
```

Comment:
Can we separate the Replayable for local and global? So that in replay_impl, we don't have to wrap them in an enum.